internal/cursorediterator/blocks.go

```diff
@@ -590,6 +587,9 @@ func CursoredProducerMapperIterator[C any, P any, I any](
 
 				select {
 				case <-ctx.Done():
+					// If we cancel, we wait for the taskrunner to complete
+					// so that there are no hanging goroutines.
+					_ = tr.Wait()
```

Comment:
This ensures that if the context is canceled, the producing goroutines are torn down before the consuming goroutine, which helps ensure that we don't get a goroutine leak.

Reply:
I'd rather not wait here; the goroutines will eventually shut down but we don't want to block the caller